sascha/iban/script.js

```diff
@@ -0,0 +1,176 @@
+        document.getElementById("copyBox").innerHTML = string;
+    }
+	
+
+//This is the function for the entire iban calculator program
+function ibanCalculator() {
+    //global variable declarations
+    var landCode = 'NL';
+    var bankRekening = document.getElementById("bankRekening").value;
+    var bankCode = document.getElementById("bankName").value;
+    var controleGetal;
+    var ibanMathString;
+	
+    /*Function Expressions*/
+    //this function resets the warning box
+    var resetWarningBox = function(){
+        document.getElementById("warningBox").className = "";
+        document.getElementById("warningBox").innerHTML = "";
+        document.getElementById("copyBox").className = "";
+        document.getElementById("copyBox").innerHTML= "";
+    }
+         //Initial checks
+	 var bankRekeningValidation = function(){
+        //if bankcode has no value ask user to choose a bank
+		if(bankCode == ""){
+           return warningMessage("Selecteer een bank a.u.b.!")
+        }
+        if(bankRekening.length < 3){
+            return warningMessage("Een geldig rekeningnummer bestaat uit minimaal 3 cijfers!");
+        }
+        if(/\D/g.test(bankRekening) === true){
+            return alert("Vul a.u.b. alleen cijfers in");
+        }
+	 }
+    //This function takes in a string and converts them in numbers according to the pattern A=10, B=11 etc.
+    var lettersToNumbers = function(string){
+        var arrToNumbers;
+        var numbers = [];
+        arrToNumbers = string.split('');
+        for(var i = 0; i < arrToNumbers.length; i++){
+            numbers.push(arrToNumbers[i].charCodeAt(0)-55);  
+        }
+        numbers = numbers.join('');
+        return numbers;
+       
+    }
+    
+    //This function grabs the account number and checks whether it's length is equal to 10, it will add zeroes in front if it does not have enough
+    var bankRekeningLengthChecker = function(acc){
+               
+        //checks whether the length of the string is equal to 10, if it is it returns the number
+        if(acc.length == 10){
+            return acc;
+        }
+        //if it's smaller than 10 it adds zeroes at the beginning so that the total input is converted to 10 numbers
+        else if(acc.length < 10){
+            //it calculates the difference between 10 and the length of the input and unshifts 0's in at the beginning based on that difference
+            var temp = 10-acc.length;
+            acc = acc.split("");
+				
+            for(var i = 0; i < temp;i++){
+                acc.unshift('0');
+            }
+            acc = acc.join("");
+            return acc;
+            }
+        else if(acc.length > 10){
+            return warningMessage("Ongeldige invoer! Vul a.u.b. een bankrekening onder de 10 cijfers in!");
+        }    
+    }
+   
+    //--------------Getters------------------------------------------------------------------------------------

```

Comment:
Netjes om je getters op deze manier te groeperen, zodat je uiteindelijk alleen deze functies hoeft aan te roepen in je applicatie.

Buiten de scope van deze opgave: je getters retourneren nu globale variabelen (globaal binnen de ibanCalculator functie), en deze waarden kunnen overal in je code aangepast en opgevraagd worden.
Verderop in het traject ga je meer object-georiënteerde code schrijven, en daar komen getters en setters pas echt tot hun recht - van een getter verwacht ik namelijk dat hij de waarde van een property van een object retourneert..